todo_app/app.py

```diff
@@ -1,13 +1,44 @@
 from flask import Flask
 from flask import render_template, request, redirect
 from todo_app.data.todo import ToDoCard, ViewModel, get_todo_cards, move_todo_card, create_todo_card
-import datetime,pytest
+import datetime, os, requests, json
+from flask_login import LoginManager, login_required, UserMixin, login_user, current_user
+from oauthlib.oauth2 import WebApplicationClient
+
+login_manager = LoginManager()
+
+@login_manager.unauthorized_handler
+def unauthenticated():
+    client = WebApplicationClient(os.environ.get('GITHUB_CLIENTID'))
```

Comment:
Don't forget to update the Heroku app with a client id and secret (a different one from your dev set because it needs the correct redirect URL)